src/azure-cli-core/azure/cli/core/adal_authentication.py

```diff
@@ -246,3 +231,27 @@ def _timestamp(dt):
     # https://docs.python.org/3/library/unittest.mock-examples.html#partial-mocking
     # https://williambert.online/2011/07/how-to-unit-testing-in-django-with-mocking-and-patching/
     return dt.timestamp()
+
+
+def aad_error_handler(error: dict):
+    """ Handle the error from AAD server returned by ADAL or MSAL. """
+    login_message = ("To re-authenticate, please {}. If the problem persists, "
+                     "please contact your tenant administrator."
+                     .format("refresh Azure Portal" if in_cloud_console() else "run `az login`"))
```

Comment:
Should we use single-quoted `'az login'` as before?

Reply:
We use backticks to mark commands, like 

https://github.com/Azure/azure-cli/blob/bba13d6f698eac8eabda83575930c0ae4d399c04/src/azure-cli-core/azure/cli/core/commands/parameters.py#L241